src/application/runner.go

```diff
@@ -67,76 +69,43 @@ func (r *Runner) getDependencyContainerNames() []string {
 	return result
 }
 
-func (r *Runner) runImages(imageNames []string, imageOnlineTexts map[string]string, identifier string) (string, error) {
-	cmdPlus, err := compose.RunImages(compose.ImagesOptions{
-		DockerComposeDir: r.DockerComposeDir,
-		ImageNames:       imageNames,
-		Logger:           r.logger,
-		Env:              []string{fmt.Sprintf("COMPOSE_PROJECT_NAME=%s", r.DockerComposeProjectName)},
+// Run runs the application with graceful shutdown
+func (r *Runner) Run() error {
+	dockerConfigs, err := composebuilder.GetApplicationDockerConfigs(composebuilder.ApplicationOptions{
+		AppConfig: r.AppConfig,
+		AppDir:    r.AppDir,
+		BuildMode: r.BuildMode,
+		HomeDir:   r.HomeDir,
 	})
 	if err != nil {
-		return cmdPlus.GetOutput(), errors.Wrap(err, fmt.Sprintf("Failed to run %s\nOutput: %s\nError: %s\n", identifier, cmdPlus.GetOutput(), err))
+		return err
 	}
-	var wg sync.WaitGroup
-	var onlineTextRegex *regexp.Regexp
-	for role, onlineText := range imageOnlineTexts {
-		wg.Add(1)
-		onlineTextRegex, err = regexp.Compile(fmt.Sprintf("%s.*%s", role, onlineText))
-		if err != nil {
-			return cmdPlus.GetOutput(), err
-		}
-		go func(role string, onlineTextRegex *regexp.Regexp) {
-			r.waitForOnlineText(cmdPlus, role, onlineTextRegex)
-			wg.Done()
-		}(role, onlineTextRegex)
+	runOptions := runner.RunOptions{
+		ImageGroups: []runner.ImageGroup{
+			{
+				ID:          "dependencies",
+				Names:       r.getDependencyContainerNames(),
```

Comment:
Should we just start calling this `getDependencyNames()` or something? Right now it seems like we're using `dependencyName`, `serviceRole`, and `ImageGroup.Name` interchangeably. I know why we do that but I feel like this could potentially get confusing if we return to this code months from now.

It seems like in this file at least they _are_ just dependency names, or even dependency keys.

Reply:
I'd like to leave this as is given that #698 will remove this.